protocols/relay/examples/docker-compose.yml

```
@@ -0,0 +1,56 @@
+version: '3.4'
+services:
+  relay: 
+    image: libp2p-relay
+    environment:
+      - relay_command_arg_env=relay
+      - deterministic_peer_env=1
+      # - address_env=/ip4/0.0.0.0/tcp/4444
+      - address_env=/ip6/2001:3984:3989::10/tcp/4444
+    build:
+      context: ../../../.
+      dockerfile: ./protocols/relay/examples/Dockerfile
+    networks:
+      default:
+        ipv4_address: 172.16.238.10
+        ipv6_address: 2001:3984:3989::10
+
+  client-listen: 
+    image: libp2p-relay
+    environment:
+      - relay_command_arg_env=client-listen
+      - deterministic_peer_env=2
+      # - address_env=/ip4/172.16.238.10/tcp/4444/p2p/12D3KooWPjceQrSwdWXPyLLeABRXmuqt69Rg3sBYbU1Nft9HyQ6X/p2p-circuit
+      - address_env=/ip6/2001:3984:3989::10/tcp/4444/p2p/12D3KooWPjceQrSwdWXPyLLeABRXmuqt69Rg3sBYbU1Nft9HyQ6X/p2p-circuit
+    build:
+      context: ../../../.
+      dockerfile: ./protocols/relay/examples/Dockerfile
+    networks:
+      default:
+        ipv4_address: 172.16.238.11
+        ipv6_address: 2001:3984:3989::11
```

```suggestion
      - address_env=/dns6/relay/tcp/4444/p2p/12D3KooWPjceQrSwdWXPyLLeABRXmuqt69Rg3sBYbU1Nft9HyQ6X/p2p-circuit
    build:
      context: ../../../.
      dockerfile: ./protocols/relay/examples/Dockerfile
    networks:
      - default
```

Why not leverage DNS?

---

I didn't succeed to do that.
When specifying it in the relay mode - address_env=/dns6/relay/tcp/4444 
I received 
Error: MultiaddrNotSupported("/dns6/relay/tcp/4444")
When specify the relay as before with:
- address_env=/ip6/::/tcp/4444
The other peers weren't able to connect to it via dns6/relay/tcp/4444 address

---

> When specifying it in the relay mode - address_env=/dns6/relay/tcp/4444 I received Error: MultiaddrNotSupported("/dns6/relay/tcp/4444")

Sorry, I forgot to mention that you will need to wrap the TCP transport with the DNS transport in order to resolve Multiaddresses with DNS addresses. The following diff should resolve your issues:


```diff
diff --git a/protocols/relay/examples/relay.rs b/protocols/relay/examples/relay.rs
index 79a3762d..a28c0d3b 100644
--- a/protocols/relay/examples/relay.rs
+++ b/protocols/relay/examples/relay.rs
@@ -54,6 +54,7 @@
 use futures::executor::block_on;
 use futures::stream::StreamExt;
 use libp2p::core::upgrade;
+use libp2p::dns::DnsConfig;
 use libp2p::ping::{Ping, PingConfig, PingEvent};
 use libp2p::plaintext;
 use libp2p::relay::{Relay, RelayConfig};
@@ -72,14 +73,14 @@ fn main() -> Result<(), Box<dyn Error>> {
     let local_peer_id = PeerId::from(local_key.public());
     println!("Local peer id: {:?}", local_peer_id);
 
-    let tcp_transport = TcpConfig::new();
+    let transport = block_on(DnsConfig::system(TcpConfig::new()))?;
 
     let relay_config = RelayConfig {
         connection_idle_timeout: Duration::from_secs(10 * 60),
         ..Default::default()
     };
     let (relay_wrapped_transport, relay_behaviour) =
-        libp2p_relay::new_transport_and_behaviour(relay_config, tcp_transport);
+        libp2p_relay::new_transport_and_behaviour(relay_config, transport);
 
     let behaviour = Behaviour {
         relay: relay_behaviour,
```

---

Your solution work. I added commit